tests/integration/internal/stepfuncs/logs.go

```diff
@@ -0,0 +1,46 @@
+package stepfuncs
+
+import (
+	"context"
+	"testing"
+
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal/logsgenerator"
+)
+
+// Generate logsCount logs using a Deployment
+func GenerateLogsWithDeployment(
+	logsCount uint,
+	logsGeneratorName string,
+	logsGeneratorNamespace string,
+	logsGeneratorImage string,
+) features.Func {
+	return func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+		client := envConf.Client()
+		generatorOptions := logsgenerator.NewDefaultGeneratorOptions()
+		generatorOptions.TotalLogs = logsCount
+		deployment := logsgenerator.GetLogsGeneratorDeployment(
+			logsGeneratorNamespace,
+			logsGeneratorName,
+			logsGeneratorImage,
+			generatorOptions,
+		)
+
+		// create the namespace
+		namespace := corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: logsGeneratorNamespace}}
+		err := client.Resources().Create(ctx, &namespace)
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
```

Comment:
```suggestion
		require.NoErrot(t, client.Resources().Create(ctx, &namespace))
```

but actually I'm not sure if this (either of those versions) wouldn't prevent the teardowns to run 🤔 can you check that?

Reply:
It's possible to stop teardowns from running by erroring out of a feature? 🙀

Reply:
`testing` framework and `e2e` framework work a bit on a different level and I'm not sure. Depends how e2e's teardown are implemented

Reply:
Ok, so the way this works is that assessments are run as subtests, so calling `t.FailNow()` (which is what both `t.Fatal()` and `require.NoError()` do) only fails the given assessment and moves on. Setups and Teardowns, however, are run in the containing test context, which means that by doing any of the aforementioned, we abort the test completely.

I'm not sure we can do any better though. If a setup or teardown action fails, we can't continue, and we don't know how to clean up after ourselves. If e2e_framework supported pairing setup and teardown actions, it could clean up correctly - maybe we should propose this feature to them?

In the interim, we either do what we're doing or roll our own finalization.

Reply:
Thanks for checking this. I'm good with the current state of affairs. I'm not sure if pairing up setups and teardowns is a necessary feature for the e2e framework 🤔 It would complicate things a bit for a debatable gain

Reply:
The gain would be the ability to clean up resources created in previous setups if one of them fails. It'd be pretty straightforward in that you'd basically pass in a second "teardown" stepfunc to Setup, and the framework would simply `defer` that stepfunc whenever it successfully completed its respective setup stepfunc. `t.FailNow` already calls all deffereds while existing.

Either way, we can merge this as-is, right?